File: Find_the_Difference.js
/**
 * @param {string} s
 * @param {string} t
 * @return {character}
 */
var findTheDifference = function(s, t) {
    const count = new Map();
    
    for (const char of t) {
        count.set(char, (count.get(char) || 0) + 1);
    }
    
    for (const char of s) {
        count.set(char, count.get(char) - 1);
        if (count.get(char) === 0) {
            count.delete(char);
        }
    }
    
    return Array.from(count.keys())[0];    
};

File: Find_the_Difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        # Initialize a dictionary to store character counts
        count = {}

        # Count characters in string t
        for c in t:
            count[c] = count.get(c, 0) + 1

        # Subtract counts for characters in string s
        for c in s:
            count[c] -= 1
            if count[c] == 0:
                del count[c]

        # The remaining character in the dictionary is the difference
        return list(count.keys())[0]
